Cesium.Preprocessor/MacroExpansionEngine.cs

```diff
@@ -10,6 +11,8 @@ namespace Cesium.Preprocessor;
 
 public class MacroExpansionEngine(IWarningProcessor warningProcessor, IMacroContext macroContext)
 {
+    private string __VA_OPT__Rule = ",";
```

Comment:
I am not quite sure about this state introduced into the engine. Note that it's reused between different preprocessor calls; it wasn't meant to have any state. And here you introduce a variable that may change while expanding `__VA_OPT__`. What if we need to expand it twice? What if we need to expand it in different macros?

I'd suggest avoid relying on this kind of state.

Reply:
Then add VA_OPT as a field in MacroArguments?